src/plone.server/plone/server/traversal.py

```diff
@@ -69,7 +130,16 @@ def __init__(self, request, resource, view, rendered):
         self.rendered = rendered
 
     async def handler(self, request):
-        view_result = await self.view()
+        if request.method in WRITING_VERBS:
+            txn = request.conn.transaction_manager.begin(request)
+            async with locked(self.resource):
+                view_result = await self.view()
+            if txn is None:
+                await sync(request)(txn.commit)
+            else:
+                await sync(request)(txn.abort)
+        else:
+            view_result = await self.view()
```

Comment:
@bloodbare I didn't get the reason, why async with for transaction manager did not work? AFAIK transaction_manager.begin should never return None. Also, the with-sugar did auto abort the transaction on all exceptions. Now there is no exception handling.